src/Mermaid.tsx

```diff
@@ -33,12 +40,40 @@ export type MermaidProps = {
  * @returns The component.
  */
 export const Mermaid = ({ chart, config }: MermaidProps): ReactElement<MermaidProps> => {
-  // init diagram once
+  // Due to Docusaurus not correctly parsing client-side from server-side modules, use the provided workaround
+  // found in the accompanying issue: https://github.com/facebook/docusaurus/issues/4268#issuecomment-783553084
+  if (typeof window === 'undefined') {
+    return <div></div>
+  }
+
+  const html: HTMLHtmlElement = document.querySelector('html')!
+
+  // Watch for changes in theme in the HTML attribute `data-theme`.
+  const [theme, setTheme] = useState<mermaidAPI.Theme>('default')
+  const observer = new MutationObserver((mutations) => {
+    for (const mutation of mutations) {
+      if (mutation.type !== 'attributes' && mutation.attributeName !== 'data-theme') {
+        continue
+      }
+
+      setTheme(html.getAttribute('data-theme') === 'dark' ? 'dark' : 'default')
```

Comment:
Can the theme used be added as config options or a callback defined in config.model.ts?

Reply:
Yup - great idea! I've added it to the config model